offload/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -205,13 +205,16 @@ Expected<std::string> getTargetTripleAndFeatures(hsa_agent_t Agent) {
       return Status;
 
     llvm::StringRef TripleTarget(ISAName.begin(), Length);
-    if (TripleTarget.consume_front("amdgcn-amd-amdhsa"))
-      Target = TripleTarget.ltrim('-').rtrim('\0').str();
-    return HSA_STATUS_INFO_BREAK;
+    if (TripleTarget.consume_front("amdgcn-amd-amdhsa")) {
+      auto Target = TripleTarget.ltrim('-').rtrim('\0').str();
+      if (Target.find("generic") != std::string::npos)
```

Comment:
`Target.ends_with("generic")` would be better.

That being said I don't think this is a good way to handle this. We should just get every isa here and then sort it later. This logic just confused me because we either assume 1. that the non-generic ISA is already first, in which case this is unnecessary, or 2 it's in any order in where we will then write the generic target, then overwrite it with the non-generic one later.

Reply:
That's why Targets[0] is reserved to obtain the specific ISA anywhere in the HSA enumeration, but I agree that wasn't a clean coding. Should I just use a vector class that supports push_front() instead of SmallVector?

Reply:
`llvm::reverse` would be equivalent, or we could just handle it without expecting an implicit order.

Reply:
If we can assume one (and only one) specific ISA and zero or more generic ISAs, we can change the incoming parameter to GenericTargets and use SpecificTarget as a return value. That may be a cleaner approach. What do you think?

Reply:
Some systems will populate with different gpu's ,  
example:  gfx906 + gfx1100  
both of these will have a specific and a generic ISA entry.

Reply:
I suppose this function is called per agent

Reply:
Yes, this code is called per agent. One potential scenario for having more than one specific ISAs enumerated per agent would be expressing backward compatibility that way. Do we need to provision the code those kinds of scenarios? Given how generic is defined today, I think that would be an overkill. Will upload the revised code.

Reply:
Revised code returns SpecificTarget ISA (same as before) and collects all ISAs w/o trying to sort it out. I believe the comments here are all addressed in that revision.